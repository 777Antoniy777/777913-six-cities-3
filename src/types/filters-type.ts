type Filter = {
  id: string,
  value: string,
};

type Filters = Filter[];
