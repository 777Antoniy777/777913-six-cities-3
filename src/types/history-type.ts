type RouteHistory = {
  action: string,
  block: () => void,
  createHref: () => void,
  go: () => void,
  goBack: () => void,
  goForward: () => void,
  length: number,
  listen: () => void,
  location: RouteLocation,
  push: () => void,
  replace: () => void,
};
