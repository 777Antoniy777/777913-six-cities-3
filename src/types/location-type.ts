type RouteLocation = {
  hash: string,
  key: string,
  pathname: string,
  search: string,
  state: string,
};
