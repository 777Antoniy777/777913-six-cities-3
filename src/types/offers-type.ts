type OfferLocation = {
  latitude: number;
  longitude: number;
  zoom: number;
};

type City = {
  name: string;
  location: OfferLocation;
};

type Host = {
  avatar: string;
  id: number;
  name: string;
  status: boolean;
};

type Offer = {
  id: number;
  city: City;
  title: string;
  premium: boolean;
  favorite: boolean;
  src: string;
  photos: string[];
  price: number;
  description: string;
  type: string;
  rating: number;
  bedroomAmount: number;
  guestsAmount: number;
  items: string[];
  host: Host;
  location: OfferLocation;
};

type Offers = Offer[];
