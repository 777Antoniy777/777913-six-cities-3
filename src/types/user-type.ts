type User = {
  id: number,
  email: string,
  name: string,
  avatar: string,
  status: boolean,
};
