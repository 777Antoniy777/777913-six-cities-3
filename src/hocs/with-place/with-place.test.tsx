import * as React from "react";
import * as renderer from "react-test-renderer";
import withPlace from "./with-place";
import {Offers} from "../../types/offers-type";
import {RouteHistory} from "../../types/history-type";
import {RouteLocation} from "../../types/location-type";

interface MockComponentProps {
  children: React.ReactNode;
}

type RouteMatch = {
  params: {
    hotelID: string;
  };
}

// set mocha data
const hockProps = {};
const offers: Offers = [
  {
    id: 1,
    city: {
      name: `city`,
      location: {
        latitude: 20,
        longitude: 20,
        zoom: 20,
      },
    },
    title: `title 1`,
    premium: false,
    favorite: false,
    src: `img/image1`,
    photos: [`img/image1`],
    price: 999999,
    description: `test`,
    type: `type`,
    rating: 9999,
    bedroomAmount: 30,
    guestsAmount: 50,
    items: [`item`],
    host: {
      id: 1,
      avatar: `img/avatar-1.jpg`,
      name: `name`,
      status: false,
    },
    location: {
      latitude: 20,
      longitude: 20,
      zoom: 20,
    },
  },
];
const location: RouteLocation = {
  hash: `hash`,
  key: `key`,
  pathname: `/pathname`,
  search: `search`,
  state: `state`,
};
const history: RouteHistory = {
  action: `action`,
  block: () => null,
  createHref: () => null,
  go: () => null,
  goBack: () => null,
  goForward: () => null,
  length: 90,
  listen: () => null,
  location,
  push: () => null,
  replace: () => null,
};
const match: RouteMatch = {
  params: {
    hotelID: `1`,
  },
};

const MockComponent = (props: MockComponentProps) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

const PlaceWrappedHoc = withPlace(MockComponent);

it(`render withPlace`, () => {
  const tree = renderer.create(
      <PlaceWrappedHoc
        props={hockProps}
        offers={offers}
        match={match}
        history={history}
        getReviews={() => null}
        getNearbyOffers={() => null}
        getActiveItem={() => null}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
