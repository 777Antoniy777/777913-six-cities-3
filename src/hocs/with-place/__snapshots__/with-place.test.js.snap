// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render withPlace 1`] = `<div />`;
